nullaway/src/main/java/com/uber/nullaway/handlers/InferredJARModelsHandler.java

```diff
@@ -68,25 +64,21 @@ private static void LOG(boolean cond, String tag, String msg) {
   }
 
   private static final int VERSION_0_FILE_MAGIC_NUMBER = 691458791;
-  private static final String DEFAULT_ASTUBX_LOCATION = "META-INF/nullaway/jarinfer.astubx";
   private static final String ANDROID_ASTUBX_LOCATION = "jarinfer.astubx";
   private static final String ANDROID_MODEL_CLASS =
       "com.uber.nullaway.jarinfer.AndroidJarInferModels";
 
   private static final int RETURN = -1; // '-1' indexes Return type in the Annotation Cache
 
   private final Map<String, Map<String, Map<Integer, Set<String>>>> argAnnotCache;
-  private final Map<String, Set<String>> mapModelJarLocations;
-  private final Set<String> loadedJars;
 
   private final Config config;
 
   public InferredJARModelsHandler(Config config) {
     super();
     this.config = config;
     argAnnotCache = new LinkedHashMap<>();
-    mapModelJarLocations = new LinkedHashMap<>();
-    loadedJars = new LinkedHashSet<>();
+    loadStubxFiles();
     // Load Android SDK JarInfer models
     try {
       InputStream androidStubxIS =
```

Comment:
@lazaroclapp do we want to keep this alternate method for loading stubs for Android?  Or should we switch our Android artifacts to also use the service loader approach?

Reply:
> Happy to switch the Android SKD models to the service loader approach (otherwise, this path is only ever used for the test project right now). It would require no configuration changes, right? Just make sure that the next release of NullAway has a models jar using auto-service and that it is still included in the classpath?

Yes, we'd need to update all the `android-jarinfer-models` artifacts to have an implementation of the new service.

> Maybe worth making that a follow up PR, so worst case it can be reverted independently?

Yes, definitely!  I'm not even sure how generation of those jars works TBH.

> Also, do we have tests for that (the Android test project maybe?) or is the only simple/existing way to test this to do so during internal upgrade at Uber?

We don't have any tests of loading android jar models in open source right now, AFAIK.